src/Tizen.Sensor/Tizen.Sensor/SensorRecorder.cs

```diff
@@ -0,0 +1,504 @@
+            if (QueryHandle != IntPtr.Zero)
+            {
+                result = true;
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Sensor Recorder Query is null");
+                throw new ArgumentException("Invalid Parameter: Sensor Recorder Query is null");
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Creates a recorder option handle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        private void CreateOption()
+        {
+            IntPtr handle;
+            int error = Interop.SensorRecoder.CreateOption(out handle);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error creating the sensor recorder option");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.CreateOption Failed");
+            }
+
+            OptionHandle = handle;
+        }
+
+        /// <summary>
+        /// Creates a recorder query handle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        private void CreateQuery()
+        {
+            IntPtr handle;
+            int error = Interop.SensorRecoder.CreateQuery(out handle);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error creating the sensor recorder query");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.CreateQuery Failed");
+            }
+
+            QueryHandle = handle;
+        }
+
+        /// <summary>
+        /// Gets a double value from a record data.
+        /// </summary>
+        /// <param name="data">
+        /// Record data handle.
+        /// </param>
+        /// <param name="key">
+        /// Data attribute to retrieve.
+        /// </param>
+        /// <returns>
+        /// returns a double value from a record data.
+        /// </returns>
+        /// <since_tizen> 6 </since_tizen>
+        public double DataGetDouble(int data, int key)
+        {
+            double value;
+            int error = Interop.SensorRecoder.DataGetDouble(data, key, out value);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error getting the sensor recorder double data");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.DataGetDouble Failed");
+            }
+            return value;
+        }
+
+        /// <summary>
+        /// Gets the start and the end time of the time period of a given record data.
+        /// </summary>
+        /// <param name="data">
+        /// Record data handle.
+        /// </param>
+        /// <returns>
+        /// returns the start and the end time of the time period of a given record data.
+        /// </returns>
+        /// <since_tizen> 6 </since_tizen>
+        public Tuple<long, long> DataGetTime(int data)
```

Comment:
Do not use Tuple, Can't know means of each values
Use struct

Reply:
Why data handle is int type?

Reply:
data handle is void* in C API. In C# API it's just used to send back that value to C API again. Internally it's being typecasted.

Reply:
Usually, void* transfer to IntPtr in C#, but we do not recommend directly exposing a IntPtr.
You should provider a wrapper class to represent a data handle